Ginger/GingerCore/Drivers/Appium/SeleniumAppiumDriver.cs

```diff
@@ -475,64 +529,72 @@ private DriverOptions GetCapabilities()
-
-                    //Tested device capabilities
-                    driverOptions.AddAdditionalCapability("deviceName", DeviceName);
-                    driverOptions.AddAdditionalCapability("platformVersion", DevicePlatformVersion);//Mobile OS version 
-                    if (!string.IsNullOrEmpty(DeviceID))
-                        driverOptions.AddAdditionalCapability("udid", DeviceID);
-
                     //Tested application capabilities 
                     driverOptions.AddAdditionalCapability("browserName", "safari");
                     //capabilities.setCapability("safariInitialUrl", "");  //(Sim-only) (>= 8.1) Initial safari url, default is a local welcome page
                     driverOptions.AddAdditionalCapability("safariAllowPopups", false);  //(Sim-only) Prevent Safari from showing a fraudulent website warning. Default keeps current sim setting.
                     //capabilities.setCapability("safariOpenLinksInBackground", false);  //(Sim-only) Whether Safari should allow links to open in new windows. Default keeps current sim setting.
-                    if (!string.IsNullOrEmpty(InstalledAppBundleID))
-                        driverOptions.AddAdditionalCapability("bundleId", InstalledAppBundleID);
                     break;
             }
 
+            //Generic capabilities
+            //driverOptions.AddAdditionalCapability("newCommandTimeout", SERVER_TIMEOUT_SEC); //How long (in seconds) Appium will wait for a new command from the client before assuming the client quit and ending the session
+            driverOptions.AddAdditionalCapability("newCommandTimeout", NewCommandTimeout.ToString());            
+            if (string.IsNullOrEmpty(AutomationName) == false)
+            {
+                driverOptions.AddAdditionalCapability("automationName", AutomationName);
+            }
+
+            //User customized capabilities
+            foreach (DriverConfigParam UserCapability in AdvanceDriverConfigurations)
+            {
+                bool boolValue;
+                int intValue=0;
+                if (bool.TryParse(UserCapability.Value, out boolValue))
+                {
+                    driverOptions.AddAdditionalCapability(UserCapability.Parameter, boolValue);
+                }
+                else if (int.TryParse(UserCapability.Value, out intValue))
+                {
+                    driverOptions.AddAdditionalCapability(UserCapability.Parameter, intValue);
+                }
+                else if(UserCapability.Value.Contains("{"))
+                {
+                    try
+                    {
+                        JObject json = JObject.Parse(UserCapability.Value);
+                        driverOptions.AddAdditionalCapability(UserCapability.Parameter, json);//for Json value to work properly, need to convert it into specific object type like: json.ToObject<selector>());
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6113827)